### ISSUE TITLE
copyto fails to copy empty directory. Related to #1837.

### ISSUE THREAD

**Issue by ipod825** [rclone/rclone#2629]
#### What is your rclone version (output from `rclone version`)
```bash
$ rclone --version
rclone v1.43
- os/arch: linux/amd64
- go version: go1.11
```

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```bash
$ mkdir -p src/a
$ mkdir dst
$ rclone copyto src/a dst/a
$ ls dst
$
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```bash
rclone -vv copyto src/a dst/a
2018/10/06 18:24:22 DEBUG : rclone: Version "v1.43.1" starting with parameters ["rclone" "-vv" "copyto" "src/a" "dst/a"]
2018/10/06 18:24:22 DEBUG : Using config file from "/home/mingo/.config/rclone/rclone.conf"
2018/10/06 18:24:22 INFO  : Local file system at /tmp/dst/a: Waiting for checks to finish
2018/10/06 18:24:22 INFO  : Local file system at /tmp/dst/a: Waiting for transfers to finish
2018/10/06 18:24:22 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 0, -
Elapsed time:       200ms

2018/10/06 18:24:22 DEBUG : 3 go routines active
2018/10/06 18:24:22 DEBUG : rclone: Version "v1.43.1" finishing with parameters ["rclone" "-vv" "copyto" "src/a" "dst/a"]

```

**Comment by ivandeex** [comment]
This issue needs more thinking.

**Comment by ivandeex** [comment]
@ncw 
Since it's known for 3+ years we could just document the behaviour, right?

**Comment by ivandeex** [comment]
@ipod825 @ishuah Does the problem reproduce with rclone 1.54?

**Comment by ipod825** [comment]
The initial test case now pass. Seems to be solved.

**Comment by ivandeex** [comment]
@ipod825 Thank you and good luck

**Comment by ipod825** [comment]
Sorry, I tried it today again and found it's still not fixed
```bash
mkdir -p src/a
mkdir dst
rclone copyto src/a dst/a
ls dst
# nothings shows 

rclone -vv copyto src/a dst/a
2021/02/12 11:05:02 DEBUG : rclone: Version "v1.54.0" starting with parameters ["rclone" "-vv" "copyto" "src/a" "dst/a"]
2021/02/12 11:05:02 DEBUG : Creating backend with remote "src/a"
2021/02/12 11:05:02 DEBUG : Using config file from "/home/smwang/.config/rclone/rclone.conf"
2021/02/12 11:05:02 DEBUG : fs cache: renaming cache item "src/a" to be canonical "/home/smwang/src/a"
2021/02/12 11:05:02 DEBUG : Creating backend with remote "dst/a"
2021/02/12 11:05:02 DEBUG : fs cache: renaming cache item "dst/a" to be canonical "/home/smwang/dst/a"
2021/02/12 11:05:02 DEBUG : Local file system at /home/smwang/dst/a: Waiting for checks to finish
2021/02/12 11:05:02 DEBUG : Local file system at /home/smwang/dst/a: Waiting for transfers to finish
2021/02/12 11:05:02 INFO  : There was nothing to transfer
2021/02/12 11:05:02 INFO  : 
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Elapsed time:         0.0s

2021/02/12 11:05:02 DEBUG : 2 go routines active

```

**Comment by ivandeex** [comment]
Sorry, my bad. I should have looked in the docs. Now I can see it explicitly documented in https://rclone.org/commands/rclone_copy/

> Note that it is always the _contents_ of the directory that is synced, _not the directory_ so when source:path is a directory, it's the contents of source:path that are copied, _not the directory name and contents_

So that's expected.

I admit, this behavior deviates from traditional `cp`. Unlike `cp`, rclone grew as a cloud-first tool where directories are frequently are only a part of _file_ name, not a real container.

We will not change that for compatibility with a wealth of rclone wrappers out there. Instead, we will add a new `rclone cp` command (see #4777) with more traditional semantics.

I think this request too can be merged into #4777

cc @ncw

**Comment by ncw** [comment]
> I think this request too can be merged into #4777

Yes, I'd agree.

**Comment by ivandeex** [comment]
Merged into https://github.com/rclone/rclone/issues/4777#issuecomment-778454797
